src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/UserRolesTest.java

```diff
@@ -64,6 +65,35 @@ public void rdcc_720_ac1_super_user_can_have_fpla_or_iac_roles() {
 
     }
 
+    @Test
+    public void rdcc_1387_ac1_super_user_can_have_caa_roles() {
+
+        String email = randomAlphabetic(10) + "@usersearch.test".toLowerCase();
+        UserCreationRequest superUser = createSuperUser(email);
+
+        professionalApiClient.getMultipleAuthHeadersExternal(puiUserManager, firstName, lastName, email);
+
+        OrganisationCreationRequest request = someMinimalOrganisationRequest()
+                .superUser(superUser)
+                .build();
+
+        Map<String, Object> response = professionalApiClient.createOrganisation(request);
+        orgIdentifier = (String) response.get("organisationIdentifier");
+        request.setStatus("ACTIVE");
+        professionalApiClient.updateOrganisation(request, hmctsAdmin, orgIdentifier);
+
+        Map<String, Object> searchUserResponse = professionalApiClient.searchUsersByOrganisation(orgIdentifier, hmctsAdmin, "false", HttpStatus.OK);
+        validateRetrievedUsers(searchUserResponse, "any");
+
+        List<Map> users = getNestedValue(searchUserResponse, "users");
+        Map superUserDetails = users.get(0);
+        List<String> superUserRoles = getNestedValue(superUserDetails, "roles");
+
+        assertThat(superUserRoles).doesNotContain("pui-caa");
+        assertThat(superUserRoles).doesNotContain("caseworker-caa");
+
```

Comment:
As we discussed this should have code to check if the roles are disabled or not.. if roles are disabled it should assert and exit from test case
If roles are enabled then it should validate the roles are added to super user

Reply:
This will be done in [RDCC-1388](https://tools.hmcts.net/jira/browse/RDCC-1388)